src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -1168,15 +1169,57 @@ impl CatalogState {
                 ServiceProcessMetrics {
                     cpu_nano_cores,
                     memory_bytes,
+                    disk_usage_bytes,
+                    disk_capacity_bytes,
                 },
             )| {
+                // We only populate a `disk_usage` if we have all 5 of of:
+                // - a disk limit (so it must be an actual managed cluster with a real limit)
+                // - a reported disk capacity
+                // - a reported disk usage
+                // - disk capacity is less than disk_usage.
+                // - There are no overflows
+                //
+                //
+                // The `disk_usage` is augmented with the difference between
+                // the limit and the `capacity`, so users don't erroneously think they
+                // have more real usable capacity than they actually do. The difference
+                // comes from LVM and filesystem overheads.
+                let disk_usage =
+                    if let (Some(disk_usage), Some(disk_capacity), Some(DiskLimit(disk_limit))) =
+                        (disk_usage_bytes, disk_capacity_bytes, disk_limit)
+                    {
+                        if disk_limit.0 >= *disk_capacity {
+                            if let Some(disk_usage) =
+                                (disk_limit.0 - *disk_capacity).checked_add(*disk_usage)
+                            {
+                                Datum::UInt64(disk_usage)
+                            } else {
+                                tracing::error!(
+                                    "disk usage  {} + correction {} overflowed?",
+                                    disk_usage,
+                                    disk_limit.0 - *disk_capacity
+                                );
+                                Datum::Null
+                            }
+                        } else {
+                            tracing::error!(
+                                "disk capacity {} is larger than the disk limit {} ?",
+                                disk_capacity,
+                                disk_limit.0
+                            );
+                            Datum::Null
+                        }
+                    } else {
+                        Datum::Null
+                    };
```

Comment:
Might be missing something, but this logic looks to be in the wrong spot to me! The mismatch between disk usage and capacity is a bug (or limitation, I suppose) of our Kubernetes orchestrator. So it feels like this logic should live in there! 

Separately, I think that'll just be way, way easier to plumb? You've already got the disk limit in the Kubernetes orchestrator, so you won't need to jump through all these hoops to get the disk limit plumbed into this method.

Was there a compelling reason to do the correction here instead?

Reply:
yeah this should be in the orchestrator! This will clean up some other code as well

Reply:
the new pushed branch has reflected this change!